instrumentation-genai/opentelemetry-instrumentation-langchain/src/opentelemetry/instrumentation/langchain/__init__.py

```diff
@@ -0,0 +1,215 @@
+                wrapper=_OpenAITraceInjectionWrapper(otel_callback_handler),
+            )
+            wrap_function_wrapper(
+                module="langchain_openai.chat_models.base",
+                name="BaseChatOpenAI._agenerate",
+                wrapper=_OpenAITraceInjectionWrapper(otel_callback_handler),
+            )
+
+    def _uninstrument(self, **kwargs):
+        """
+        Cleanup instrumentation (unwrap).
+        """
+        unwrap("langchain_core.callbacks.base", "BaseCallbackManager.__init__")
+        if not self._disable_trace_injection:
+            # unwrap("langchain_community.llms.openai", "BaseOpenAI._generate")
+            # unwrap("langchain_community.llms.openai", "BaseOpenAI._agenerate")
+            # unwrap("langchain_community.llms.openai", "BaseOpenAI._stream")
+            # unwrap("langchain_community.llms.openai", "BaseOpenAI._astream")
+
+            unwrap("langchain_openai.llms.base", "BaseOpenAI._generate")
+            unwrap("langchain_openai.llms.base", "BaseOpenAI._agenerate")
+            unwrap("langchain_openai.llms.base", "BaseOpenAI._stream")
+            unwrap("langchain_openai.llms.base", "BaseOpenAI._astream")
+
+            unwrap("langchain_openai.chat_models.base", "BaseChatOpenAI._generate")
+            unwrap("langchain_openai.chat_models.base", "BaseChatOpenAI._agenerate")
+
+
+class _BaseCallbackManagerInitWrapper:
+    """
+    Wrap the BaseCallbackManager __init__ so we can insert
+    our custom callback handler in the manager's handlers list.
+    """
+
+    def __init__(self, callback_handler):
+        self._otel_handler = callback_handler
+
+    def __call__(self, wrapped, instance, args, kwargs):
+        wrapped(*args, **kwargs)
+        # Ensure our OTel callback is present if not already.
+        for handler in instance.inheritable_handlers:
+            if isinstance(handler, type(self._otel_handler)):
+                break
+        else:
+            instance.add_handler(self._otel_handler, inherit=True)
+
+
+class _OpenAITraceInjectionWrapper:
+    """
+    A wrapper that intercepts calls to the underlying LLM code in LangChain
+    to inject W3C trace headers into upstream requests (if possible).
+    """
+
+    def __init__(self, callback_manager):
+        self._otel_handler = callback_manager
+
+    def __call__(self, wrapped, instance, args, kwargs):
+        """
+        We look up the run_id in the `kwargs["run_manager"]` to find
+        the active span from the callback handler. Then we inject
+        that span context into the 'extra_headers' for the openai call.
+        """
+        run_manager = kwargs.get("run_manager")
+        if run_manager is not None:
+            run_id = run_manager.run_id
+            span_holder = self._otel_handler.spans.get(run_id)
+            if span_holder and span_holder.span.is_recording():
+                extra_headers = kwargs.get("extra_headers", {})
+                context = span_holder.span_context
```

Comment:
Looks like the context value that had been previously saved is not used. Should it be passed into the `set_span_in_context` call?